terraform/deploy

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+
+set -eu
+
+if [ $# -lt 1 ]; then
+    echo "usage: $0 [ENVIRONMENT] [COMMAND ...]"
+    exit 1
+fi
+
+ENV=${1:-staging}
+shift
+
+conf=$(dirname $0)/deploy.conf
+if [ -r "$conf" ]; then
+    . "$conf"
+fi
+if [ -z "${aws_profile:-}" ] || [ -z "${aws_mfa_serial:-}" ]; then
+    echo "E: Both \$aws_profile and \$aws_mfa_serial must be set."
+    echo "I: Either set them in the environment, or in $conf"
+    exit 1
+fi
+
+# data
+role=arn:aws:iam::495570029050:role/qa-admin
+duration=3600
+
+# assume the qa-admin role
+if [ -e .admin_credentials ]; then
+    now=$(date +%s)
+    credentials_created=$(stat --format=%Y .admin_credentials)
+    if [ $((now - credentials_created)) -gt 3600 ]; then
+        rm -f .admin_credentials
+    fi
+fi
+if [ ! -e .admin_credentials ] || [ ! -s .admin_credentials ]; then
+    read -p "MFA token: " token
+    touch .admin_credentials
+    chmod 600 .admin_credentials
+    aws --profile "$aws_profile" \
+        --region us-east-1 \
+        sts assume-role \
+        --role-arn "$role"  \
+        --role-session-name "`whoami`" \
+        --duration-seconds "$duration" \
+        --query 'Credentials.[AccessKeyId,SecretAccessKey,SessionToken]' \
+        --serial-number "$aws_mfa_serial" \
+        --token-code "$token" \
+        --output text \
+        >> .admin_credentials
+fi
+
+export AWS_ACCESS_KEY_ID="$(awk '{print($1)}' .admin_credentials)"
+export AWS_SECRET_ACCESS_KEY="$(awk '{print($2)}' .admin_credentials)"
+export AWS_SESSION_TOKEN="$(awk '{print($3)}' .admin_credentials)"
+
+# environment
+./scripts/set_env.sh "$ENV"
+eval $(./scripts/load_db_password "$ENV")
+
+# init if needed
+if [ ! -d .terraform ]; then
+    terraform init
+fi
+
+if [ $# -eq 0 ]; then
+    # deploy
+    terraform apply -var-file=shared.tfvars -var-file=${ENV}.tfvars .
+
+    # set the stage for ansible
+    terraform state pull | ./scripts/state_to_ansible_inventory.py > "../ansible/hosts.${ENV}"
+else
+    "$@"
+fi
```

Comment:
A few general comments/concerns on this file:
- Let's not have both a Makefile and a deploy file, if they serve the same purpose. 
- I guess this file's interface could use some README or similar. i.e. i would have expected 'terraform plan' to be the default (read-only operation)
- It's called 'deploy' but really it's just a local terraform wrapper that supports any terraform operation